tket/test/src/test_CompilerPass.cpp

```diff
@@ -1722,5 +1722,114 @@ SCENARIO("Flatten and relabel registers") {
   }
 }
 
+SCENARIO("Custom rebase pass with implicit wire swaps.") {
+  OpTypeSet allowed_gates_cx = {OpType::PhasedX, OpType::Rz, OpType::CX};
+  PassPtr pp_rebase_cx = gen_rebase_pass_via_tk2(
+      allowed_gates_cx, CircPool::TK2_using_CX_and_swap,
+      CircPool::tk1_to_PhasedXRz);
+  OpTypeSet allowed_gates_zzmax = {OpType::PhasedX, OpType::Rz, OpType::ZZMax};
+  PassPtr pp_rebase_zzmax = gen_rebase_pass_via_tk2(
+      allowed_gates_zzmax, CircPool::TK2_using_ZZMax_and_swap,
+      CircPool::tk1_to_PhasedXRz);
+
+  GIVEN("Targeting CX gates, ISWAPMax gate.") {
+    Circuit c(2);
+    c.add_op<unsigned>(OpType::ISWAPMax, {0, 1});
+    CompilationUnit cu(c);
+    CHECK(pp_rebase_cx->apply(cu));
+    REQUIRE(cu.get_circ_ref().count_gates(OpType::CX) == 1);
```

Comment:
Can we add checks of unitary equivalence to all these tests please?

Reply:
Have done -> for some reason Sycamore is not providing equivalent unitaries, I will need to investigate

Reply:
I'm not sure why I didn't check this first, but after some bug hunting I have found that actually the issue is not with the new code, but in the top level `standard_rebase_via_tk2`, which even with `CircPool::TK2_using_CX` instead of the swap version, produces incorrect unitary matrices.

My suggestion is to add a new jira task to fix this separately, and push with the Sycamore tests commented out.